### DISCUSSION TITLE
I'm new to React, how come I get [object Object] all the time?

### DISCUSSION THREAD

**Discussion by indigofficial** [community/community#21526] [Other Feature Feedback, Questions, & Ideas]
<p>Hi!</p>
<p>So recently I was studying React, on a website project. I don’t have the github files yet, as an error occurred, but here is my <strong>counter.jsx</strong> file:</p>
<pre><code>import React, { Component } from "react";
import { throwStatement, thisTypeAnnotation } from "@babel/types";

class Counter extends Component {
  state = {
    count: 0
  };

handleIncrement = () =&gt; {
    this.setState({ count: this.state + 1 });
  };

render() {
    return (
      &lt;div&gt;
        &lt;span className={this.getBadgeClasses()}&gt;{this.formatCount()}&lt;/span&gt;
        &lt;button
          onClick={this.handleIncrement}
          className="btn btn-secondary btn-sm"
        &gt;
          Increment
        &lt;/button&gt;
&lt;/div&gt;
    );
  }

getBadgeClasses() {
    let classes = "badge m-2 badge-";
    classes += this.state.count === 0 ? "warning" : "primary";
    return classes;
  }

  formatCount() {
    const { count } = this.state;
    return count === 0 ? "Zero" : count;
  }
}

export default Counter;
</code></pre>
<p>And my <strong>index.js</strong> file:</p>
<pre><code>import React from "react";
import ReactDOM from "react-dom";
import "./index.css";
import App from "./App";
import * as serviceWorker from "./serviceWorker";
import "bootstrap/dist/css/bootstrap.css";
import Counter from "./components/counter";

ReactDOM.render(&lt;Counter /&gt;, document.getElementById("root"));

serviceWorker.unregister();
</code></pre>
<p>So the problem is, whenever I click the Increment button, I always get:</p>
<p><img src="https://user-images.githubusercontent.com/82490573/181065104-6ee1dd5e-0f8e-4725-b30c-bc138dbb1b8b.png" alt="Capture.PNG" data-base62-sha1="u1HKSk7DVzpi6AJQ0qs0ob8P4fm" width="365" height="78"></p>
<p>How do I get rid of the [object Object]1?</p>
<p>Thank you,</p>
<ul>
<li>Zoe K.</li>
</ul>

**Comment by badsyntax** [comment, marked as answer]
<p>I’m on mobile so apologies for lack of formatting. You are incrementing the state object instead of increment the count property:</p>
<p>Your code: this.state + 1<br>
Should be: this.state.count + 1</p>

**Comment by indigofficial** [comment]
<p>Thank you <a class="mention" href="/badsyntax">@badsyntax</a>! It helped for me!!</p>
